router/h2/src/main/scala/io/buoyant/router/h2/ForwardClientCertFilter.scala

```diff
@@ -0,0 +1,66 @@
+package io.buoyant.router.h2
+
+import java.security.MessageDigest
+import java.security.cert.X509Certificate
+import javax.xml.bind.DatatypeConverter.printHexBinary
+
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.h2.{Request, Response}
+import com.twitter.finagle.transport.Transport
+import com.twitter.util.Future
+
+import scala.collection.mutable
+
+class ForwardClientCertFilter extends SimpleFilter[Request, Response] {
+
+  val GeneralNameTypeUri = 6
+
+  def apply(req: Request, svc: Service[Request, Response]): Future[Response] = {
+    Transport.peerCertificate.foreach(clientCert => {
+      val clientCertHeader = new mutable.StringBuilder(128)
+
+      val digest = MessageDigest.getInstance("SHA-256")
+      clientCertHeader ++= s"Hash=${printHexBinary(digest.digest(clientCert.getEncoded))}"
+
+      clientCert match {
+        case x509ClientCert: X509Certificate =>
```

Comment:
is this match exhaustive?  don't you need to handle the case where the cert is not an X509Certificate to avoid a match error?